features/ubuntu_pro_fips.feature

```diff
@@ -0,0 +1,152 @@
+        Then stdout matches regexp:
+            """
+            .*esm-apps      +yes +enabled +UA Apps: Extended Security Maintenance \(ESM\)
+            esm-infra     +yes +enabled +UA Infra: Extended Security Maintenance \(ESM\)
+            fips          +yes +<fips-s> +NIST-certified core packages
+            fips-updates  +yes +disabled +NIST-certified core packages with priority security updates
+            livepatch     +yes +n/a  +Canonical Livepatch service.*
+            """
+        When I run `cat /var/log/squid/access.log` `with sudo` on the `proxy` machine
+        Then stdout matches regexp:
+        """
+        .*CONNECT contracts.canonical.com.*
+        """
+        And stdout does not match regexp:
+        """
+        .*CONNECT 169.254.169.254.*
+        """
+        Examples: ubuntu release
+           | release | fips-s   |
+           | xenial  | enabled  |
+           | bionic  | enabled  |
+           | focal   | enabled  |
+
+    @series.lts
+    @uses.config.machine_type.aws.pro.fips
+    Scenario Outline: Check fips is enabled correctly on Ubuntu pro fips AWS machine
+        Given a `<release>` machine with ubuntu-advantage-tools installed
+        When I create the file `/etc/ubuntu-advantage/uaclient.conf` with the following:
+        """
+        contract_url: 'https://contracts.canonical.com'
+        data_dir: /var/lib/ubuntu-advantage
+        log_level: debug
+        log_file: /var/log/ubuntu-advantage.log
+        """
+        And I run `ua auto-attach` with sudo
+        And I run `ua status --wait` as non-root
+        And I run `ua status` as non-root
+        Then stdout matches regexp:
+            """
+            .*esm-apps      +yes +enabled +UA Apps: Extended Security Maintenance \(ESM\)
+            esm-infra     +yes +enabled +UA Infra: Extended Security Maintenance \(ESM\)
+            fips          +yes +<fips-s> +NIST-certified core packages
+            fips-updates  +yes +disabled +NIST-certified core packages with priority security updates
+            livepatch     +yes +n/a  +Canonical Livepatch service.*
+            """
+        And I verify that running `apt update` `with sudo` exits `0`
+        And I verify that running `grep Traceback /var/log/ubuntu-advantage.log` `with sudo` exits `1`
+        And I verify that `openssh-server` is installed from apt source `<fips-apt-source>`
+        And I verify that `openssh-client` is installed from apt source `<fips-apt-source>`
+        And I verify that `strongswan` is installed from apt source `<fips-apt-source>`
+        And I verify that `openssh-server-hmac` is installed from apt source `<fips-apt-source>`
+        And I verify that `openssh-client-hmac` is installed from apt source `<fips-apt-source>`
+        And I verify that `strongswan-hmac` is installed from apt source `<fips-apt-source>`
+        When I run `uname -r` as non-root
+        Then stdout matches regexp:
+            """
+            aws-fips
```

Comment:
On xenial, It seems like the `ubuntu-fips` metapackage is being installed, so we will not see the `aws-fips` string here, only `fips`

Reply:
Okay, I noticed I have been testing on bionic mostly. Let me do the tests on xenial as well. Just having issues with updating the python versions

Reply:
So, technical the string ".*-fips" should work

Reply:
Yes, this does work, but I think we should be explicti that `bionic` expected the optimized kernel while xenial does not. We can add a new column on the examples table to say the expected kernel version

Reply:
I agree with @lucasmoura we want to test that the correct specific kernel is installed in each version.